packages/utils/src/types.ts

```diff
@@ -0,0 +1,5 @@
+// Opaque numeric types to avoid mixing different denominations of bitcoin
+type BitcoinUnit<K, T> = K & { _: T };
+export type Btc = BitcoinUnit<number, 'Btc'>;
+export type Sats = BitcoinUnit<number, 'Sats'>;
+export type MSats = BitcoinUnit<number, 'MSats'>;
```

Comment:
Should we use bigint for sats and/or mSats?

Reply:
Sorry I should have written some notes about why I removed your `bigint` types and made them `number`. Using `bigint` isn't as simple as changing the type since it's an entirely separate class from `Number` and has runtime differences. JSON parsing of numbers will not automatically cast them as `BigInt`, they will initially be `Number`s which means we would have to do the class conversion ourselves, which means that we'd already be suffering from the accuracy loss at that point. The Rust codebase would have to serialize them as strings instead of numbers in order for us to preserve accuracy, and that would mean quite a few breaking changes.

I talked to Justin about this, but we both decided that it would probably be overkill. Here's why:
```ts
formatMsatsToBtc(Number.MAX_SAFE_INTEGER)
// '90,071.99254741'
```
You would need to have 90k bitcoin before you would start to see some numeric inaccuracies in your balances. While this could conceivably happen in a regtest environment, this is unlikely to ever happen even on testnet / mutinynet since it's hard to accrue that many coins. Even if that did happen, JavaScript continues to function with numbers well over the max "safe" integer, it just starts to introduce inaccuracies in the lower digits. So you'd start to see some rounding at the satoshi level. But we're not going to get any runtime crashes, and we _probably_ won't end up with crazy numbers.

Might be worth adding a note somewhere, I've made the mistake of going all strings / big numbers for bitcoin stuff under this impression in the past but it's just practically not worth the headache. But lmk if you have other concerns.